File: app/src/main/java/cn/com/szw/lib/myframework/base/MyBaseFragment.java
package cn.com.szw.lib.myframework.base;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.View;
import android.view.ViewGroup;

import cn.com.szw.lib.myframework.utils.RxBus;
import cn.com.szw.lib.myframework.view.CustomProgress;


public abstract class MyBaseFragment extends Fragment {
	protected View rootView;
	@Override
	public void onDestroy() {
		try {
			RxBus.get().unregister(this);
		}catch (Exception e){
			e.printStackTrace();
		}
		CustomProgress.disMiss();
		super.onDestroy();
	}
	@Override
	public void onDestroyView() {
		super.onDestroyView();
		if (rootView != null) {
			ViewGroup view = (ViewGroup) rootView.getParent();
			if (view!=null)
			view.removeView(rootView);
		}
	}
	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		initView();
		initEvent();
		try {
			RxBus.get().register(this);
		}catch (Exception e){
			e.printStackTrace();
		}
	}
	public  abstract void initView();
	public  void initEvent(){}
}

File: Flibrary/src/main/java/com/szw/framelibrary/base/MyBaseFragment.kt
package com.szw.framelibrary.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View
import android.view.ViewGroup

import com.hwangjr.rxbus.RxBus
import com.szw.framelibrary.view.CustomProgress


 abstract class MyBaseFragment : Fragment() {
    lateinit var rootView:View
    override fun onDestroy() {
        try {
            RxBus.get().unregister(this)
        } catch (e: Exception) {
            e.printStackTrace()
        }

        CustomProgress.disMiss()
        super.onDestroy()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        val view = rootView.parent as ViewGroup
        view.removeView(rootView)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        initEvent()
        try {
            RxBus.get().register(this)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    abstract fun initView()
    fun initEvent() {}
}
